test/e2e/gatewayapi_upgradeable_test.go

```diff
@@ -0,0 +1,94 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"k8s.io/client-go/kubernetes"
+	"testing"
+
+	operatorcontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	gatewayapi_upgradeable "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/gatewayapi-upgradeable"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func TestGatewayAPIUpgradeable(t *testing.T) {
+	t.Parallel()
+
+	t.Log("Starting test: Ensure the Gateway API CRDs exist")
+	ensureCRDs(t)
+
+	t.Log("Starting test: Testing if admin gate works properly")
+	testAdminGate(t)
+
+	t.Log("Starting test: Deleting Gateway API CRDs")
+	deleteCRDs(t)
+
+	t.Log("Starting test: Testing if admin gate works properly after CRDs are deleted")
+	testAdminGate(t)
+
+	// Defer the cleanup of the test gateway.
+	t.Cleanup(func() {
+		t.Log("Starting cleanup: Deleting test gateway")
+		testGateway := gatewayapiv1.Gateway{ObjectMeta: metav1.ObjectMeta{Name: testGatewayName, Namespace: operatorcontroller.DefaultOperandNamespace}}
+		if err := kclient.Delete(context.TODO(), &testGateway); err != nil {
+			if errors.IsNotFound(err) {
+				t.Logf("Gateway %q not found during cleanup", testGateway.Name)
+				return
+			}
+			t.Errorf("failed to delete gateway %q: %v", testGateway.Name, err)
+		} else {
+			t.Logf("Successfully deleted gateway %q", testGateway.Name)
+		}
+	})
+}
+
+func getClient() (*kubernetes.Clientset, error) {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(kubeConfig)
+}
+
+func testAdminGate(t *testing.T) {
+	kclient, err := getClient()
+	if err != nil {
+		t.Fatalf("failed to get kube client: %v", err)
+	}
+
+	t.Log("Ensuring CRDs exist")
+	crdsExist := true
+
+	t.Logf("Getting configmap admin-gates in namespace openshift-config-managed")
+	configMap, err := kclient.CoreV1().ConfigMaps("openshift-config-managed").
+		Get(context.TODO(), "admin-gates", metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("failed to get configmap admin-gates: %v", err)
+		if errors.IsNotFound(err) {
+			t.Logf("ConfigMap 'admin-gates' not found in namespace openshift-config-managed")
+		} else {
+			t.Logf("Error getting ConfigMap: %v", err)
+		}
+		return
+	}
+
+	t.Log("Checking for admin gate key in the configmap")
+	adminGateKeyExists := false
+	if _, ok := configMap.Data[gatewayapi_upgradeable.GatewayAPIAdminKey]; ok {
+		adminGateKeyExists = true
+	}
+
+	t.Logf("CRDs exist: %v, Admin Gate Key exists: %v", crdsExist, adminGateKeyExists)
```

Comment:
This is ok to log while you debug your test but, it's too verbose for a job which will be run as a pre-submit for every PR.